decide-app/src/main/java/com/decide/app/LaunchInterceptorConditions.java

```diff
@@ -56,8 +56,36 @@ public boolean getLaunchInterceptorCondition1() {
         return true;
     }
 
+    /**
+     * Launch Interceptor Condition 2:
+     * There exists at least one set of three consecutive data points
+     * which form an angle such that:
+     * angle < (PI−EPSILON) or angle > (PI+EPSILON)
+     * The second of the three consecutive points is always the vertex of the angle. 
+     * If a point coincides with the vertex, the angle is undefined and the LIC is not satisfied.
+     * @return True if the condition is met, false otherwise.
+     */
     public boolean getLaunchInterceptorCondition2() {
-        return true;
+        if (PARAMETERS.EPSILON < 0 || PARAMETERS.EPSILON >= Math.PI) throw new IllegalArgumentException();
+        double a;
+        Point vertex;
+        double x1;
+        double x2;
+        double y1;
+        double y2;
+        for(int i = 0; i < NUMPOINTS-2; i++){
+            vertex = POINTS[i+1];
+            if(POINTS[i].getX() == vertex.getX() && POINTS[i].getY() == vertex.getY()
+            || POINTS[i+2].getX() == vertex.getX() && POINTS[i+2].getY() == vertex.getY()) return false;
+            x1 = (POINTS[i].getX() - vertex.getX());
+            y1 = (POINTS[i].getY() - vertex.getY());
+            x2 = (POINTS[i+2].getX() - vertex.getX());
+            y2 = (POINTS[i+2].getY() - vertex.getY());
+            a = Math.abs(Math.atan2(y2,x2) - Math.atan2(y1, x1));

```

Comment:
I think this is correct, but I would preferably like to see a source (here in the comments)

Reply:
Found it here: https://stackoverflow.com/questions/7066792/angle-between-3-points-signed-bad-results 

and here: https://stackoverflow.com/questions/1211212/how-to-calculate-an-angle-from-three-points